lib/util.js

```diff
@@ -434,7 +434,85 @@ async function* streamAsyncIterator(stream) {
   }
 }
 
+/**
+ * Implements the Blob interface but the data is loaded lazily when needed from
+ * a source URL provided on creation.
+ */
+class LazyFile {
+  #blob = null;
+  #source = null;
+  #name = null;
+
+  constructor(source, options) {
+    if (source instanceof URL) {
+      this.#source = source;
+    } else {
+      this.#blob = new Blob(source, options);
+    }
+
+    this.#name = options?.name;
+  }
+
+  toString() {
+    return `[LazyFile ${this.#source ?? this.#blob.toString()}]`;
+  }
```

Comment:
So far, we've been conservative about JS language features like class fields, optional chaining, and nullish coalescing. These aren't new by any measure, but I'm still surprised to see them causing problems in the wild. To the extent that we can do the same without syntactic sugar (e.g. `||` instead of `??`, `&&` instead of `?.`), I think that'd be preferable.

Reply:
I think we need to tighten up how we talk about features by specifying platforms that we support and versions. We've made great progress here with the integration suite.

Currently all newer JS runtimes support these language features, Deno, Bun, CloudFlare etc, and Node has supported it since v14 released in 2020, we currently only work in 18+. All browsers since 2020 support these features (and our library doesn't work in the browser).

I'm all for being conservative, but I do also think we need to be pragmatic.